### PR TITLE
feat(cli): add command

### PR DESCRIPTION
**Pull request by alicanerdurmaz**
feat: added `add` command to add new features to the project
Now you can add a new provider or resource to the project using the command `npm run refine add <arg>`.

-   `npm run refine add resource posts`: will add a new resource to the project with the name `posts`.
-   `npm run refine add auth`: will add a new auth provider to the project.
-   `npm run refine add data`: will add a new data provider to the project.
-   `npm run refine add live`: will add a new live provider to the project.

### Self Check before Merge

Please check all items below before review.

-   [x] Corresponding issues are created/updated or not needed
-   [x] Docs are updated/provided or not needed
-   [x] Examples are updated/provided or not needed
-   [x] TypeScript definitions are updated/provided or not needed
-   [x] Tests are updated/provided or not needed
-   [x] Changesets are provided or not needed

### CODE REVIEW

File: packages/cli/templates/provider/demo-data-provider.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        console.log("getList", {
+            resource,
+            pagination,
+            filters,
+            sorters,
+            meta,
+            url,
+        });
+
+        // TODO: send request to the API
```
Comment (BatuhanW) [inline packages/cli/templates/provider/demo-data-provider.tsx:19]: Let's add relevant documentation links to all templates.

File: packages/cli/templates/provider/demo-auth-provider.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import { AuthBindings } from "@refinedev/core";
+
+export const authProvider: AuthBindings = {
```
Comment (aliemir) [inline packages/cli/templates/provider/demo-auth-provider.tsx:3]:
```
/**
 * Check out the Auth Provider documentation for detailed information
 * https://refine.dev/docs/api-reference/core/providers/auth-provider/
**/
```

Maybe we can add something like this in the demo providers 🤔 @alicanerdurmaz

**Review by aliemir — APPROVED** [review]

File: documentation/docs/packages/documentation/cli.md [hunk 3]
```diff
@@ -158,17 +159,56 @@ If there is already a file with the same name in the directory, the swizzle comm
...
+| resourceName (optional) | The name of the resource you want to add. |
+
+#### provider
+
+Empty Provider methods are created for to easily implement and connect to your data source from your **refine** project. These components are put on the specified path.
+
+```bash
+> npm run refine add auth
+> npm run refine add data
+> npm run refine add live
```
Comment (omeraplak) [inline documentation/docs/packages/documentation/cli.md:185]:
```suggestion
> npm run refine add live
> npm run refine add i18n
> npm run refine add notification
> npm run refine add accessControl
```
Can we update it for our other options as well?

**Review by omeraplak — CHANGES_REQUESTED** [review]

**Comment by omeraplak** [comment]
Hey @alicanerdurmaz,
Can we add a test?

**Review by omeraplak — APPROVED** [review]

**Review by BatuhanW — APPROVED** [review]

**Pull request merged**